DMApp.Keystone/AppPackageCreatorForKeystone.cs

```diff
@@ -0,0 +1,100 @@
+﻿namespace Skyline.DataMiner.CICD.DMApp.Keystone
+{
+    using System;
+    using System.Diagnostics;
+    using System.Threading.Tasks;
+
+    using Skyline.DataMiner.CICD.DMApp.Common;
+    using Skyline.DataMiner.CICD.FileSystem;
+    using Skyline.DataMiner.CICD.Loggers;
+
+    using static Skyline.AppInstaller.AppPackage;
+
+    /// <summary>
+    /// Represents a creator for application packages specifically for Keystone dashboards within the DataMiner System.
+    /// </summary>
+    public class AppPackageCreatorForKeystone : AppPackageCreator
+    {
+        private readonly ToolMetaData toolMetaData;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AppPackageCreatorForKeystone"/> class.
+        /// </summary>
+        /// <param name="toolMetaData">Metadata associated with the tool.</param>
+        /// <param name="fileSystem">File system interface to manage file operations.</param>
+        /// <param name="logCollector">Log collector to capture logs during operations.</param>
+        /// <param name="directoryPath">The directory path where packages are to be created.</param>
+        /// <param name="packageName">The name of the package.</param>
+        /// <param name="packageVersion">The version of the package.</param>
+        public AppPackageCreatorForKeystone(ToolMetaData toolMetaData, IFileSystem fileSystem, ILogCollector logCollector, string directoryPath, string packageName, DMAppVersion packageVersion) : base(fileSystem, logCollector, directoryPath, packageName, packageVersion)
+        {
+            this.toolMetaData = toolMetaData;
+        }
+
+        /// <summary>
+        /// Asynchronously adds items to the application package.
+        /// </summary>
+        /// <param name="appPackageBuilder">The application package builder to which the items are added.</param>
+        /// <returns>A task that represents the asynchronous operation.</returns>
+        public override async Task AddItemsAsync(AppPackageBuilder appPackageBuilder)
+        {
+            if (appPackageBuilder == null) throw new ArgumentNullException(nameof(appPackageBuilder));
+            string pathToCreatedTool;
+
+            if (RepositoryPath.EndsWith(".nupkg"))
```

Comment:
When the path is the nupkg file itself, it fails already on the constructor as it expects a directory: 
![image](https://github.com/SkylineCommunications/Skyline.DataMiner.CICD.Packages/assets/97458010/8005ad77-d801-4e40-9af2-25c66552be4c)

Reply:
Fixed this.